snabble/Android-SDK
# Improve/refactor use of API desugaring

The SDK already depends on the default desugaring variant. When switching to the nio variant, we won't need the manual for moveFile anymore. The apk file size gains an additional ~100 KB _(82.2 MB total)_, which is reasonable to change simply upgrade to the nio variant.

Further more, the desugaring library has been added to any of the modules, which seems unnecessary. It's been removed where it seemed to have no use. Desugaring is required because of:
* `:core` `Coupon.kt` by _java.time.ZonedDateTime_,
* `:ui` `CouponItem.kt` by _java.time.chrono.ChronoZonedDateTime#toEpochSecond_
* `:utils` `StringDownloader.java` by _org.apache.commons.io.IOUtils_

The app integrating the Sabble SDK has to enable api desugaring anyways - because of that is has been added to the README.

### How to test?
Integrate the SDK using this branch in an app and deploy it on an e.g. Android 6 _(API 23)_ device. It should crash on launch, if no desugaring with nio as stated in the readme has been added. If the correct nio API desugaring is added to the app, it should work.

### Definition of Done

#### App Tests
